dnf-behave-tests/features/fail-safe-cache.feature

```diff
@@ -0,0 +1,308 @@
+#         - nodejs:5 - nodejs-1:5.3.1-1.module_2011+41787af0.x86_64
+#         - nodejs:8 - nodejs-1:8.11.4-1.module_2030+42747d40.x86_64
+#         - nodejs:10 - nodejs-1:10.11.0-1.module_2200+adbac02b.x86_64
+#         - nodejs:11 - nodejs-1:11.0.0-1.module_2311+8d497411.x86_64
+#     dnf-ci-fedora-modular-updates:
+#         nodejs-1:10.14.1-1.module_2533+7361f245.x86_64
+#         nodejs-1:11.1.0-1.module_2379+8d497405.x86_64
+#         nodejs-1:12.1.0-1.module_2379+8d497405.x86_64
+#         nodejs-1:8.11.4-1.module_2030+42747d40.x86_64
+#         nodejs-1:8.14.0-1.module_2030+42747d40.x86_64
+#         - nodejs:8 - nodejs-1:8.11.4-1.module_2030+42747d40.x86_64
+#         - nodejs:10 - nodejs-1:10.14.1-1.module_2533+7361f245.x86_64
+#         - nodejs:11 - nodejs-1:11.1.0-1.module_2379+8d497405.x86_64
+#         - nodejs:12 - nodejs-1:12.1.0-1.module_2379+8d497405.x86_64
+
+
+@bz1616167
+@bz1623128
+Scenario: Fail-safe modulemd copy is created for every enabled stream and nothing else
+   Then file "/var/lib/dnf/modulefailsafe/" does not exist
+   When I execute dnf with args "module enable nodejs:5 postgresql:10"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                    | Package                |
+        | module-stream-enable      | nodejs:5               |
+        | module-stream-enable      | postgresql:10          |
+    And modules state is following
+        | Module         | State     | Stream    | Profiles  |
+        | nodejs         | enabled   | 5         |           |
+        | postgresql     | enabled   | 10        |           |
+   Then file "/var/lib/dnf/modulefailsafe/" exists
+   When I execute "ls" with args "{context.dnf.installroot}/var/lib/dnf/modulefailsafe/"
+   Then stdout is
+        """
+        nodejs:5:x86_64.yaml
+        postgresql:10:x86_64.yaml
+        """
+
+
+@bz1616167
+@bz1623128
+Scenario: Fail-safe modulemd copy is created ONLY for the stream of the highest available version and correctly updated
+   Then file "/var/lib/dnf/modulefailsafe/" does not exist
+   When I execute dnf with args "module enable nodejs:8"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                    | Package                |
+        | module-stream-enable      | nodejs:8               |
+    And modules state is following
+        | Module         | State     | Stream    | Profiles  |
+        | nodejs         | enabled   | 8         |           |
+   Then file "/var/lib/dnf/modulefailsafe/" exists
+   When I execute "ls" with args "{context.dnf.installroot}/var/lib/dnf/modulefailsafe/"
+   Then stdout is
+        """
+        nodejs:8:x86_64.yaml
+        """
+    And file "/var/lib/dnf/modulefailsafe/nodejs:8:x86_64.yaml" contains lines
+        """
+        version: 20181216123422
+        """
+    And file "/var/lib/dnf/modulefailsafe/nodejs:8:x86_64.yaml" does not contain lines
+        """
+        version: 20180816123422
```

Comment:
Probably it is better to use versions that are easily distinguishable. Those versions are nearly identical.

Reply:
I changed the version of nodejs:8 in dnf-ci-fedora-modular, so that it is better distinguishable.